### PR TITLE
Compile variable parsing.

### PR DESCRIPTION
**Pull request by ruiaraujo**
Once ready it will replace the `getVariableValues` from `graphql-js`

These are the current results
```
graphql-js x 4,009 ops/sec ±4.79% (77 runs sampled)
graphql-jit - default x 139,388 ops/sec ±2.63% (75 runs sampled)
graphql-jit - default no leaf x 157,109 ops/sec ±2.56% (76 runs sampled)
graphql-jit - compiled x 200,618 ops/sec ±2.68% (75 runs sampled)
graphql-jit - compiled no leaf x 247,419 ops/sec ±2.74% (74 runs sampled)
Fastest is graphql-jit - compiled no leaf
```

### CODE REVIEW

**Comment by ruiaraujo** [comment]
:+1:

**Comment by ruiaraujo** [comment]
:+1:

**Comment by ruiaraujo** [comment]
:+1:

File: src/variables.ts [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    }
+
+    body += "if (errors.length > 0) {return {errors, coerced: undefined};}";
+    body += "return {errors: undefined, coerced};\n}";
+    return Function
+        .apply(null, ["coerced", "GraphQLError", "inspect"].concat(Array.from(dependencies.keys())).concat(body))
+        .apply(null, [coercedValues, GraphQLError, inspect].concat(Array.from(dependencies.values())));
+}
+
+const MAX_INT = 2147483647;
```
Comment (topicus) [inline src/variables.ts]: Is this number special in GQL? You can get the max safe int and min safe int using Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER.

Comment (ruiaraujo) [reply src/variables.ts]: https://graphql.github.io/graphql-spec/June2018/#sec-Int

Comment (boopathi) [reply src/variables.ts]: Use `0x7fffffff`

**Comment by topicus** [comment]
:+1:

**Comment by ruiaraujo** [comment]
:+1:

**Pull request merged**